distrobuilder/chroot.go

```diff
@@ -8,7 +8,7 @@ import (
 	"github.com/lxc/distrobuilder/shared"
 )
 
-func manageRepositories(def *shared.Definition, manager *managers.Manager) error {
+func manageRepositories(def *shared.Definition, manager *managers.Manager, vmOnly bool) error {
```

Comment:
So I'm not sure about that `vmOnly` approach. What if I now pass:
```yaml:
types:
 - container
```

And want a container only change? AFAICT this wouldn't work, right?

Reply:
See [here](https://github.com/lxc/distrobuilder/pull/277/commits/7528172e2c76791abae1dfd958ce3b99b3b2e4df#diff-37c4bfabb2fb9bdb701896ec3f8b596cR521-R525)

Reply:
Ah right because we still do actual matching of target type and available types.

So looking at the other branch, `vmOnly` is set when we do pack for a VM so we only run the VM bits then. Wouldn't we need something similar for when we pack a container, in which case we'd only want the container type entries to be applied at that time?

Reply:
> Wouldn't we need something similar for when we pack a container, in which case we'd only want the container type entries to be applied at that time?

I guess we would. Currently, container-only stuff would always be installed. I'll fix that.

Reply:
I added some logic which also can handle container-only sections. The tests pass, but I still need to test it properly by actually building containers.

Reply:
@stgraber I just tested building and packing containers, and it works as expected.